subsys/bluetooth/host/classic/sdp.c

```diff
@@ -1727,13 +1905,204 @@ static void sdp_client_notify_result(struct bt_sdp_client *session,
+		return 0;
+	}
+
+	if ((record_len + SDP_CONT_STATE_LEN_SIZE + cstate->length) >
+		buf->len) {
+		LOG_ERR("Invalid payload length");
+		return 0;
+	}
+
+	/*
+	 * No record found for given UUID. The check catches case when
+	 * current response frame has Continuation State shortest and
+	 * valid and this is the first response frame as well.
+	 */
+	if (!current_count && cstate->length == 0U &&
+		session->cstate.length == 0U) {
+		LOG_DBG("record for service record handle 0x%x not found",
+			session->param->handle);
+		/* Call user UUID handler */
+		sdp_client_notify_result(session, UUID_NOT_RESOLVED);
+		net_buf_pull(buf, sizeof(cstate->length));
+		goto iterate;
+	}
+
+	if (record_len > net_buf_tailroom(session->rec_buf)) {
+		LOG_WRN("Not enough room for getting records data");
+		goto iterate;
+	}
+
+	net_buf_add_mem(session->rec_buf, buf->data, record_len);
+	net_buf_pull(buf, record_len);
+
+	/* check if current response says there's next portion to be fetched */
+	if (cstate->length) {
+		/* Cache original Continuation State in context */
+		memcpy(&session->cstate, cstate,
+			sizeof(struct bt_sdp_pdu_cstate));
+
+		net_buf_pull(buf, cstate->length +
+				sizeof(cstate->length));
+
+		/* Request for next portion of attributes data */
+		return sdp_client_discover(session);
+	}
+
+	net_buf_pull(buf, sizeof(cstate->length));
+
+	LOG_DBG("UUID 0x%s resolved", bt_uuid_str(session->param->uuid));
+	sdp_client_notify_result(session, UUID_RESOLVED);
+iterate:
+	/* Get next UUID and start resolving it */
+	sdp_client_params_iterator(session);
+
+	return 0;
+}
+
+static int sdp_client_receive_ssa_sa(struct bt_sdp_client *session, struct net_buf *buf)
+{
+	struct bt_sdp_pdu_cstate *cstate;
+	uint16_t frame_len;
+	uint16_t total;
+
+	/* Check the buffer len for the frame_len field */
+	if (buf->len < sizeof(frame_len)) {
+		LOG_ERR("Invalid frame payload length");
+		return 0;
+	}
+
+	/* Get number of attributes in this frame. */
+	frame_len = net_buf_pull_be16(buf);
+	/* Check valid buf len for attribute list and cont state */
+	if (buf->len < frame_len + SDP_CONT_STATE_LEN_SIZE) {
+		LOG_ERR("Invalid frame payload length");
+		return 0;
+	}
+	/* Check valid range of attributes length */
+	if (frame_len < 2) {
+		LOG_ERR("Invalid attributes data length");
+		return 0;
+	}
+
+	/* Get PDU continuation state */
+	cstate = (struct bt_sdp_pdu_cstate *)(buf->data + frame_len);
+
+	if (cstate->length > BT_SDP_MAX_PDU_CSTATE_LEN) {
+		LOG_ERR("Invalid SDP PDU Continuation State length %u", cstate->length);
+		return 0;
+	}
+
+	if ((frame_len + SDP_CONT_STATE_LEN_SIZE + cstate->length) >
+		buf->len) {
+		LOG_ERR("Invalid frame payload length");
+		return 0;
+	}
+
+	/*
+	 * No record found for given UUID. The check catches case when
+	 * current response frame has Continuation State shortest and
+	 * valid and this is the first response frame as well.
+	 */
+	if (frame_len == 2U && cstate->length == 0U &&
+		session->cstate.length == 0U) {
```

Comment:
Incorrectly aligned line, which makes the subsequent LOG_DBG statement indistinguishable from being part of the if-condition itself. However, you can probably just merge here too. You'll note that the clang-format checker is actually suggesting you to do exactly this.

Reply:
Yes. I updated all according to the suggestion of `clang-format checker`.